File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.entity.PageResult;
import com.pinyougou.entity.Result;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

/**
 * 品牌的控制器曾
 */
@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){
        return brandService.findAll();
    }


    @RequestMapping("/findPage")
    public PageResult findPage(int page,int rows){
        return brandService.findPage(page,rows);
    }

    /**
     * 增加
     * @param tbBrand
     * @return
     */
    @RequestMapping("/addTbBrand")
    public Result addTbBrand(@RequestBody TbBrand tbBrand){
        try{
            brandService.addTbBrand(tbBrand);
            return new Result(true,"增加成功");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"增加失败");
        }
    }

    /**
     * 修改品牌信息
     * @param tbBrand
     * @return
     */
    @RequestMapping("/updateTbBrand")
    public Result Update(@RequestBody TbBrand tbBrand){
        try{
            brandService.UpdateTbBrand(tbBrand);
            return new Result(true,"修改成功");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }

    /**
     * 删除一个或多个品牌
     * @param ids
     * @return
     */
    @RequestMapping("/deleteTbBrand")
    public Result delete(long[] ids){
        try{
            brandService.deleteTbBrand(ids);
            return new Result(true,"删除成功");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }

    /**
     * 获取实体
     * @param id
     * @return
     */
    @RequestMapping("/findOne")
    public TbBrand findOne(Long id){
        return brandService.findOne(id);
    }

    /**
     * 条件查询
     * @param tbBrand
     * @param page
     * @param rows
     * @return
     */
    @RequestMapping("/search")
    public PageResult search(@RequestBody TbBrand tbBrand, int page, int rows){
        return brandService.getAll(tbBrand,page,rows);
    }

    /**
     * 获取品牌下拉列表
     * @return
     */
    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList(){
        return brandService.selectOptionList();
    }


}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
//构建前端控制层
app.controller("brandController",function ($scope,$controller,brandService) {
    $controller("baseController",{$scope:$scope});

    $scope.findAll=function () {
        brandService.findAll().success(function (response) {
            $scope.list=response;
        })
    }

    $scope.findPage=function(page,rows){
        brandService.findPage(page,rows).success(function(response){
                $scope.list=response.rows;
                $scope.paginationConf.totalItems=response.total;//更新总记录数
            }
        );
    }

    //条件查询分页
    $scope.search=function(page,rows){
        brandService.search(page,rows,$scope.searchEntity).success(
            function (response) {
                $scope.list = response.rows;
                //定义总记录数
                $scope.paginationConf.totalItems= response.total;
            }
        )
    }

    //查询实体
    $scope.findOne=function(id){
        brandService.findOne(id).success(
            function(response){
                $scope.entity=response;
            }
        );
    }

    //保存
    $scope.save=function () {
        /*var methodName = 'addTbBrand';
        if ($scope.entity.id != null) {//如果有ID
            methodName = 'updateTbBrand';//则执行修改方法
        }
        brandService.save().success(
                function (response) {
                    if (response.success) {
                        //重新查询
                        $scope.reloadList();//重新加载
                    } else {
                        alert(response.message);
                    }
                }
        );*/
        var object = null;
        if($scope.entity.id!=null){
            object=brandService.updateTbBrand($scope.entity);
        }else{
            object=brandService.addTbBrand($scope.entity);
        }
        object.success(function (response) {
            if (response.success) {
                //重新查询
                $scope.reloadList();//重新加载
            } else {
                alert(response.message);
            }
        })


        /*if($scope.entity.id!=null){//如果有ID
            methodName='update';//则执行修改方法
        }
        $http.post('../brand/'+ methodName +'.do',$scope.entity ).success(
                function(response){
                    if(response.success){
                        //重新查询
                        $scope.reloadList();//重新加载
                    }else{
                        alert(response.message);
                    }
                }
        );*/
    }
    //删除
    $scope.dele = function () {
        //出现过删除无效的问题 原因处在这：点击复选框时并未将该id加入复选框
        if($scope.selectIds.length==0){
            alert("请选择");
            return;
        }
        //获取选中的复选框
        brandService.dele($scope.selectIds).success(function (response) {
            if (response.success) {
                $scope.reloadList();//重新加载
                $scope.selectIds = [];
            }else{
                alert(response.message);
            }
        });
    }


})
